### PR TITLE
Fixes #22 - ExecBuilder.asDaemon didn't work in some cases

### LINKED ISSUE #22: ExecBuilder.asDaemon doesn't work on EC2
Internal ID: 56

Doing trivial `ExecBuilder.fromCommand("touch", "/tmp/bagr").exec(node)` works properly, but when adding `.asDaemon()`, the file doesn't get created.

In our tests, We're using a public AMI with Amazon Linux on it. The sleep command from our tests works on that AMI when run manually, but not when running without a terminal. I don't know what's the cause yet. However, when running the tests on a RH AMI, it works in both cases. Thus, I assume the issue is with the Amazon AMI and `ExecBuilder.asDaemon` does work on EC2.

### PR DESCRIPTION
**Pull request by kwart**
Fixes #22

### CODE REVIEW

File: core/src/test/java/org/wildfly/extras/sunstone/tests/AbstractCloudProviderTest.java [hunk 1]
```diff
@@ -274,15 +274,14 @@ private void testExecBuilder(Node node) throws IOException, InterruptedException
         assertThat(node.exec("cat", "/tmp/my-sudo-id.txt").getOutput(), containsString("root"));
 
         long sleepInSecs = 60;
-        ExecBuilder sleepCheckExec = ExecBuilder.fromCommand("pgrep", "sleep");
-        assertNotEquals("No running 'sleep' process should be found", 0, sleepCheckExec.exec(node).getExitCode());
+        assertEquals("No running 'sleep' process should be found", 1, node.exec("pgrep","sleep").getExitCode());
```
Comment (Ladicek) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/AbstractCloudProviderTest.java]: IMHO in this case, `!= 0` is slightly better than `== 1`, but I don't really care :-)

**Comment by Ladicek** [comment]
If it fixes the test (which I guess it does :-) ), +1

File: core/src/main/java/org/wildfly/extras/sunstone/api/process/ExecBuilder.java [hunk 2]
```diff
@@ -110,16 +109,10 @@ public ExecResult exec(Node node) throws OperationNotSupportedException, IOExcep
...
         try (SshClient ssh = node.ssh()) {
-            ExecResult execResult;
+            ExecResult execResult = ssh.execAndWait(renderedCommand);
             if (asDaemon) {
-                try (CommandExecution commandExecution = ssh.exec(renderedCommand)) {
-                    // empty is OK, we don't want to touch the streams - it's the daemon.
-                    LOGGER.debug("SSH CommandExecution was created for running command on node '{}' as daemon: {}",
-                            node.getName(), renderedCommand);
-                }
+                LOGGER.trace("Nohup execution result (will not be propageted): {}", execResult);
```
Comment (LittleJohnII) [inline core/src/main/java/org/wildfly/extras/sunstone/api/process/ExecBuilder.java]: "propageted" -> "propagated"

**Comment by LittleJohnII** [comment]
LGTM, I'll merge it tomorrow.

**Comment by kwart** [comment]
Typo fixed. The `AzureTest` failed with following error, but it seems unrelated. I'll rerun it.

@Ladicek do you know why I don't see a full stack trace for this error? Is it due to our logging config? Or maven/surefire?

```
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 894.747 sec <<< FAILURE! - in org.wildfly.extras.sunstone.tests.azure.AzureTest
node(org.wildfly.extras.sunstone.tests.azure.AzureTest)  Time elapsed: 893.739 sec  <<< ERROR!
java.util.concurrent.CompletionException: org.jclouds.http.HttpResponseException: command: DELETE https://management.core.windows.net/2586c64b-38b4-4527-a140-012d49dfc02c/services/disks/sst-jcacek-361-sst-jcacek-361-0-201608181341580105 HTTP/1.1 failed with response: HTTP/1.1 400 Bad Request; content: [<Error xmlns="http://schemas.microsoft.com/windowsazure" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><Code>BadRequest</Code><Message>A disk with name sst-jcacek-361-sst-jcacek-361-0-201608181341580105 is currently in use by virtual machine sst-jcacek-361 running within hosted service sst-jcacek-361, deployment sst-jcacek-361.</Message></Error>]
Caused by: org.jclouds.http.HttpResponseException: command: DELETE https://management.core.windows.net/2586c64b-38b4-4527-a140-012d49dfc02c/services/disks/sst-jcacek-361-sst-jcacek-361-0-201608181341580105 HTTP/1.1 failed with response: HTTP/1.1 400 Bad Request; content: [<Error xmlns="http://schemas.microsoft.com/windowsazure" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><Code>BadRequest</Code><Message>A disk with name sst-jcacek-361-sst-jcacek-361-0-201608181341580105 is currently in use by virtual machine sst-jcacek-361 running within hosted service sst-jcacek-361, deployment sst-jcacek-361.</Message></Error>]
```

**Comment by LittleJohnII** [comment]
OK, the test fail really seems like some Azure quirk. Merging.

**Pull request merged**